Todomvc-Aurelia 0.1.0

mhoyer opened the pull request:
This is an attempt to create the TodoMVC app with Aurelia.

## Pros

- fully TDD based development => 70 specs for release 0.1.0
- manually verified [TodoMVC App Specification](https://github.com/tastejs/todomvc/blob/master/app-spec.md) requirements
- development followed mainly the approaches of 
  - http://aurelia.io/get-started.html 
  - https://github.com/aurelia/skeleton-navigation
  - https://github.com/SAAirey/aurelia-skeleton-navigation-bundling

## "Cons"

Some things that still need to be discussed/considered:

- The original TodoMVC browser-test suite is currently not passing (at least on my Win7x64 machine).
  - See https://github.com/tastejs/todomvc/issues/1147
  - In turn, I see the same error when trying to run the reference implementation (backbone example) from latest todomvc repository. 
- The bundling and export of dependencies and actual application code doesn't feel mature (but it's automated through `gulp build`). 
  - How to provide an instant view for the user without flickering on page load? This is a somehow fundamental issue as SystemJS is recommended module loader for Auriala, which implies asynchronous loading of all dependencies and the app (incl. views) itself.
- Aurelia is still in pre-beta state. So changes in the API need to be applied for this sample application - what I'll try to do from time to time. ;-)

sindresorhus commented on examples/aurelia/README.md:
```diff
@@ -0,0 +1,62 @@
+
+## Implementation
+
+- following the getting started tutorial: http://aurelia.io/get-started.html
```
Capitalize the first character

sindresorhus commented on examples/aurelia/README.md:
```diff
@@ -0,0 +1,62 @@
+#### 0.1.0
+
+- finished functional implementation to fulfill [TodoMVC Application Specification](https://github.com/tastejs/todomvc/blob/master/app-spec.md)
+- includes bundling (jspm) and exporting (for easy sync with [todomvc/examples/aurelia](https://github.com/tastejs/todomvc/tree/master/examples/aurelia))
```
Capitalize the first character

sindresorhus commented on examples/aurelia/config.js:
```diff
@@ -0,0 +1,383 @@
+System.config({
+  "paths": {
```
Don't quote keys and use single-quotes. This isn't JSON.

sindresorhus commented on examples/aurelia/css/app.css:
```diff
@@ -0,0 +1 @@
+/* all comes from todomvc-app-css package */
```
This can be removed.

sindresorhus commented on examples/aurelia/dist/todos.html:
```diff
@@ -0,0 +1,48 @@
+<template>
+  <import from="./behaviors/focus"></import>
```
`./` is moot.

sindresorhus commented:
Found a couple of bugs:

- The 'All' filter should be active on load. (the one at the bottom)
- The app is missing the left "learn bar". See the other apps.

mhoyer commented:
@sindresorhus thanks for good feedback. I'll apply your findings possibly later this day.

mhoyer commented:
@sindresorhus I just added a squashed commit containing the fixes for the issues you pointed out. Were you able to let the browser-test suite run against the Aurelia example on your side? Or anyone else?

mhoyer commented:
What's the status on this PR? Do you need any more support from my side?

sindresorhus commented:
I'm fine with landing us landing this when the below is fixed and commits squashed. @tastejs/todomvc ?

- Do you really have to include these files? Seems more like dev dependencies.

```
examples/aurelia/jspm_packages/traceur.src.js
examples/aurelia/jspm_packages/es6-module-loader.src.js
examples/aurelia/jspm_packages/system.src.js
```

- Bug: Double-click to edit doesn't work when in a route (filter), like `Active`.

- Can you also fill out https://github.com/tastejs/todomvc/blob/master/learn.json for Aurelia

mhoyer commented:
@sindresorhus, I was able to remove those ```.src.js``` files, fixed the double-click bug, filled the ```learn.json```, and I squashed everything to a single commit.

If any further support is needed, please let me know.

addyosmani commented:
I'm :+1: on us landing this too. Confirmed the learn.json was populated as needed and the other bugs appear to have been addressed. Deferring to @sindresorhus for a merge when he's happy.

sindresorhus commented:
Landed! Awesome work on getting this together @mhoyer :)

![tumblr_mkwooez15j1qdlh1io1_400](https://cloud.githubusercontent.com/assets/170270/6851638/514d3f60-d41b-11e4-98f7-ce03b530d40b.gif)

arthurvr commented:
W00t! Awesome work @mhoyer!
